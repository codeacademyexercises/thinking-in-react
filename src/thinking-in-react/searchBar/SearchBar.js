import React, { Component } from 'react';
import '../App/App.css';

class SearchBar extends Component{
    render(){
        return(
            <div className="SearchBarContainer">
                <div>
                    <input type="search" placeholder="Search..." id="searchBar" defaultValue={this.props.search} onChange={this.props.SearchInput}/>
                </div>
                <div>
                    <input type="checkbox" id="inStock" defaultValue={this.props.stock} onChange={this.props.InStock}/>
                    <label >Only show products in stock</label>
                </div>
                
            </div>
        );
    }
}

export default SearchBar;