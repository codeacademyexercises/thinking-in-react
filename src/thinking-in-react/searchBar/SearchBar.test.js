import React from 'react';
import SearchBar from './SearchBar.js';
import Renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

describe('Checking ',()=>{
    it('Snapshot test will work',()=>{
        const tree = Renderer.create(<SearchBar/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('count of each ClassName is correct and Text beside checkbox is as specified',()=>{
        const wrapper = shallow(<SearchBar/>);
        expect(wrapper.find('.SearchBarContainer')).toHaveLength(1);
        expect(wrapper.text()).toEqual("Only show products in stock");
    });
    it('onChange functions to be called on change',()=>{
        let wrapper = shallow(<SearchBar/>);
        const checkbox = jest.fn();
        const searchbar = jest.fn();
        wrapper = wrapper.setProps({
            search: 'Electronics',
            stock: true,
            InStock: checkbox,
            SearchInput: searchbar,
        });
        wrapper.find('input').at(1).simulate('change');
        expect(checkbox).toHaveBeenCalled();
        wrapper.find('input').at(0).simulate('change');
        expect(searchbar).toHaveBeenCalled();
    });
    it('defaultValues are set as specified',()=>{
        const wrapper = shallow(<SearchBar stock={true} search="Electronics"/>);
        expect(wrapper.find('input').at(0).props().defaultValue).toEqual("Electronics");
        expect(wrapper.find('input').at(1).props().defaultValue).toEqual(true);
    });
});