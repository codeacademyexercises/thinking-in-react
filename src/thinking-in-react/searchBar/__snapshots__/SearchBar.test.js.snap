// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking  Snapshot test will work 1`] = `
<div
  className="SearchBarContainer"
>
  <div>
    <input
      id="searchBar"
      placeholder="Search..."
      type="search"
    />
  </div>
  <div>
    <input
      id="inStock"
      type="checkbox"
    />
    <label>
      Only show products in stock
    </label>
  </div>
</div>
`;
